run_linter.sh

```diff
@@ -0,0 +1,76 @@
+#!/bin/sh
+#
+# Copyright (C) 2019 Ultimaker B.V.
+#
+# SPDX-License-Identifier: LGPL-3.0+
+
+set -eu
+
+DOCKER_WORK_DIR="${DOCKER_WORK_DIR:-/build}"
+SHELLCHECK_ARGS="-x -C -f tty"
+
+usage()
+{
+    echo "Usage: ${0} [OPTIONS]"
+    echo "Run this repository's linter"
+    echo "  -h   Print usage"
+}
+
+run_in_shell()
+{
+    eval "${@}"
+}
+
+run_shellcheck()
+{
+    if command -v "docker" 1> /dev/null; then
+        SHELLCHECK_CMD="docker run \
+            --rm \
+            -v \"$(pwd):${DOCKER_WORK_DIR}\" \
+            -w \"${DOCKER_WORK_DIR}\" \
+            registry.hub.docker.com/koalaman/shellcheck:stable"
+    else
+        echo "You need Docker to run the shellcheck linter."
+        return
+    fi
+
+    SCRIPTS="$(find "./scripts/" "./test/" -name '*.sh')"
+    for shellcheck_script in "./"*".sh"  "./legacy_test/"*".sh" ${SCRIPTS}; do
```

Comment:
One small thing on the commandline i found the following warning: Warning, skipping shellcheck './legacy_test/*.sh'
legacy_tests is not something we want to copy, please remove this.